### PR TITLE
change docker entrypoint/cmd annotations to json

### PR DESCRIPTION
**Pull request by sjpotter**
can't serialize them with a join as exec arguments can have a space

### CODE REVIEW

**Comment by sjpotter** [comment]
@yifan-gu

**Comment by yifan-gu** [comment]
LGTM

**Comment by yifan-gu** [comment]
@iaguis

**Comment by iaguis** [comment]
Can you prefix the commit with "lib/internal:"?

This is just so the users can inspect how the original image was, right?

**Comment by iaguis** [comment]
> This is just so the users can inspect how the original image was, right?

I'll answer myself: yes (I just checked the original PR for these annotations).

LGTM after changing the commit message

File: lib/internal/internal.go [hunk 1]
```diff
@@ -182,10 +182,18 @@ func GenerateManifest(layerData types.DockerImageData, dockerURL *types.ParsedDo
...
+                       }
+                       annotations = append(annotations, appctypes.Annotation{Name: *appctypes.MustACIdentifier(common.AppcDockerEntrypoint), Value: string(entry)})
 		}
 		if len(dockerConfig.Cmd) > 0 {
-			annotations = append(annotations, appctypes.Annotation{Name: *appctypes.MustACIdentifier(common.AppcDockerCmd), Value: strings.Join(dockerConfig.Cmd, " ")})
+                       cmd, err := json.Marshal(dockerConfig.Cmd)
+                       if err != nil {
+                               return nil, err
+                       }
+                       annotations = append(annotations, appctypes.Annotation{Name: *appctypes.MustACIdentifier(common.AppcDockerCmd), Value: string(cmd)})
```
Comment (iaguis) [inline lib/internal/internal.go]: Weird, I wonder why `Value` is colored here and not above...

Comment (yifan-gu) [inline lib/internal/internal.go]: @sjpotter I think you didn't gofmt it.

**Pull request merged**